userspace/libsinsp/threadinfo.cpp

```diff
@@ -511,88 +514,82 @@ void sinsp_threadinfo::init(scap_threadinfo* pi) {
 	}
 }
 
 void sinsp_threadinfo::set_group(uint32_t gid) {
+	m_gid = gid;
 	scap_groupinfo* group = m_inspector->m_usergroup_manager.get_group(m_container_id, gid);
 	if(!group) {
 		auto notify = m_inspector->is_live() || m_inspector->is_syscall_plugin();
-		group = m_inspector->m_usergroup_manager.add_group(m_container_id, m_pid, gid, {}, notify);
+		m_inspector->m_usergroup_manager.add_group(m_container_id, m_pid, gid, {}, notify);
 	}
-	if(group) {
-		m_group.set_gid(group->gid);
-
-		if(m_inspector->is_user_details_enabled()) {
-			m_group.set_name(group->name, strnlen(group->name, MAX_CREDENTIALS_STR_LEN));
-		}
-	} else {
-		// No need to set name/homedir/shell, the default values from
-		// sinsp_userinfo are going to be used.
-		m_group.set_gid(gid);
-	}
-	m_user.set_gid(m_group.gid());
 }
 
 void sinsp_threadinfo::set_loginuser(uint32_t loginuid) {
-	scap_userinfo* login_user = m_inspector->m_usergroup_manager.get_user(m_container_id, loginuid);
-
-	if(login_user) {
-		m_loginuser.set_uid(login_user->uid);
-		m_loginuser.set_gid(m_group.gid());
-
-		if(m_inspector->is_user_details_enabled()) {
-			m_loginuser.set_name(login_user->name,
-			                     strnlen(login_user->name, MAX_CREDENTIALS_STR_LEN));
-			m_loginuser.set_homedir(login_user->homedir,
-			                        strnlen(login_user->homedir, MAX_CREDENTIALS_STR_LEN));
-			m_loginuser.set_shell(login_user->shell,
-			                      strnlen(login_user->shell, MAX_CREDENTIALS_STR_LEN));
-		}
-	} else {
-		// No need to set name/homedir/shell, the default values from
-		// sinsp_userinfo are going to be used.
-		m_loginuser.set_uid(loginuid);
-		m_loginuser.set_gid(m_group.gid());
-	}
-}
-
-const sinsp_threadinfo::cgroups_t& sinsp_threadinfo::cgroups() const {
-	return m_cgroups;
+	m_loginuid = loginuid;
 }
 
-std::string sinsp_threadinfo::get_comm() const {
-	return m_comm;
+scap_userinfo* sinsp_threadinfo::get_user() const {
+	auto user = m_inspector->m_usergroup_manager.get_user(m_container_id, m_uid);
+	if(user != nullptr) {
+		return user;
+	}
+	static scap_userinfo usr{};
+	usr.uid = m_uid;
+	usr.gid = m_gid;
+	strlcpy(usr.name, m_uid == 0 ? "root" : "<NA>", sizeof(usr.name));
+	strlcpy(usr.homedir, m_uid == 0 ? "/root" : "<NA>", sizeof(usr.homedir));
+	strlcpy(usr.shell, "<NA>", sizeof(usr.shell));
+	return &usr;
 }
 
-std::string sinsp_threadinfo::get_exe() const {
-	return m_exe;
+scap_groupinfo* sinsp_threadinfo::get_group() const {
+	auto group = m_inspector->m_usergroup_manager.get_group(m_container_id, m_gid);
+	if(group != nullptr) {
+		return group;
+	}
+	static scap_groupinfo grp = {};
+	grp.gid = m_gid;
+	strlcpy(grp.name, m_gid == 0 ? "root" : "<NA>", sizeof(grp.name));
+	return &grp;
 }
 
-std::string sinsp_threadinfo::get_exepath() const {
-	return m_exepath;
+scap_userinfo* sinsp_threadinfo::get_loginuser() const {
+	auto user = m_inspector->m_usergroup_manager.get_user(m_container_id, m_loginuid);
```

Comment:
this is the same code we use in `set_user` apart from a different id, can we create an helper method